cca.go

```diff
@@ -0,0 +1,118 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat
+
+import (
+	"errors"
+	"math"
+
+	"github.com/gonum/floats"
+	"github.com/gonum/matrix"
+	"github.com/gonum/matrix/mat64"
+)
+
+// CanonicalCorrelations returns the canonical correlations, corr,
+// between the corresponding pairs of canonical variables. Canonical
+// variables can be obtained by projecting the sphered data into the
```

Comment:
Same with canonical variables and canonical correlation matrix, I'm not sure how to fix this other than to add a sentence explaining that these ideas will be introduced below.